ir/inst_conversion.go

```diff
@@ -536,6 +553,12 @@ type InstBitCast struct {
 // NewBitCast returns a new bitcast instruction based on the given source value
 // and target type.
 func NewBitCast(from value.Value, to types.Type) *InstBitCast {
+	if _, isFromPtr := from.Type().(*types.PointerType); !isFromPtr {
```

Comment:
I'm fairly sure you can bitcase, e.g. a 64-bit integer into a 64-bit floating point value.

> llvm/test/CodeGen/Hexagon/float-bitcast.ll:  `%v0 = bitcast double %a0 to i64`